app/models/peer_review.rb

```diff
@@ -8,9 +10,23 @@ class PeerReview < ActiveRecord::Base
   validates_presence_of :result
   validates_numericality_of :reviewer_id, only_integer: true, greater_than: 0
   validates_numericality_of :result_id, only_integer: true, greater_than: 0
+  validate :no_students_should_be_reviewer_and_reviewee
 
   def reviewee
     # TODO - Research optimizing or see if rails can do better
     Grouping.joins({ submissions: { results: :peer_reviews }}).where('peer_reviews.id = ?', self.id).first
   end
+
+  def no_students_should_be_reviewer_and_reviewee
+    if result and reviewer
```

Comment:
It is surprising that this check is needed if the validations already do this.

Perhaps the related objects in the tests are simply invalid. You should check the factories.